Mathlib/Algebra/Order/Sub/Defs.lean

```diff
@@ -0,0 +1,452 @@
+/-
+Copyright (c) 2021 Floris van Doorn. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Floris van Doorn
+-/
+import Mathlib.Algebra.CovariantAndContravariant
+import Mathlib.Algebra.Group.Basic
+import Mathlib.Algebra.Order.Monoid.Lemmas
+import Mathlib.Order.Lattice
+
+/-!
+# Ordered Subtraction
+
+This file proves lemmas relating (truncated) subtraction with an order. We provide a class
+`HasOrderedSub` stating that `a - b ≤ c ↔ a ≤ c + b`.
+
+The subtraction discussed here could both be normal subtraction in an additive group or truncated
+subtraction on a canonically ordered monoid (`ℕ`, `Multiset`, `PartEnat`, `Ennreal`, ...)
+
+## Implementation details
+
+`HasOrderedSub` is a mixin type-class, so that we can use the results in this file even in cases
+where we don't have a `CanonicallyOrderedAddMonoid` instance
+(even though that is our main focus). Conversely, this means we can use
+`CanonicallyOrderedAddMonoid` without necessarily having to define a subtraction.
+
+The results in this file are ordered by the type-class assumption needed to prove it.
+This means that similar results might not be close to each other. Furthermore, we don't prove
+implications if a bi-implication can be proven under the same assumptions.
+
+Lemmas using this class are named using `tsub` instead of `sub` (short for "truncated subtraction").
+This is to avoid naming conflicts with similar lemmas about ordered groups.
+
+We provide a second version of most results that require `[ContravariantClass α α (+) (≤)]`. In the
+second version we replace this type-class assumption by explicit `AddLeCancellable` assumptions.
```

Comment:
```suggestion
second version we replace this type-class assumption by explicit `AddLECancellable` assumptions.
```

Reply:
This disagrees with Algebra.Order.Monoid.Lemmas. Should that file also be changed?